src/test/java/de/unijena/cheminf/mortar/model/io/DynamicSMILESFileReaderTest.java

```diff
@@ -0,0 +1,210 @@
+        Assertions.assertEquals(1, tmpMolSet.getAtomContainerCount());
+        Assertions.assertEquals("CNP0356547", tmpMolSet.getAtomContainer(0).getProperty(Importer.MOLECULE_NAME_PROPERTY_KEY));
+
+        /*
+        Expected output:    2 parsable lines
+                            2 invalid lines
+        Test file's specifications:
+        - headline and blank line first
+        - three elements per line
+        - SMILES first in line, ID second and a neglectable third element
+        - third element in line
+        - used separator: "\t"
+         */
+        tmpURL = this.getClass().getResource("SMILESTestFileFive.txt");
+        tmpFormat = DynamicSMILESFileReader.detectFormat(Paths.get(tmpURL.toURI()).toFile());
+        Assertions.assertTrue(tmpFormat.hasHeaderLine());
+        Assertions.assertEquals(1, tmpFormat.getIDColumnPosition());
+        Assertions.assertEquals(0, tmpFormat.getSMILESCodeColumnPosition());
+        Assertions.assertTrue(tmpFormat.hasIDColumn());
+        Assertions.assertEquals('\t', tmpFormat.getSeparatorChar());
+        tmpMolSet = DynamicSMILESFileReader.readFile(Paths.get(tmpURL.toURI()).toFile(), tmpFormat);
+        //
+        String[] tmpTestFileFiveSmiles = new String[] {"OC=1C=C(O)C=C(C1)C=2OC=3C=CC=CC3C2", "OC=1C=C(O)C(=C(C1)C(C)C(O)C)C"};
+        String[] tmpTestFileFiveIDs = new String[] {"CNP0192622", "CNP0262448"};
+        int i = 0;
+        for (IAtomContainer tmpAtomContainer :
+                tmpMolSet.atomContainers()) {
+            Assertions.assertEquals(tmpTestFileFiveSmiles[i],ChemUtil.createUniqueSmiles(tmpAtomContainer));
+            Assertions.assertEquals(tmpTestFileFiveIDs[i],tmpAtomContainer.getProperty(Importer.MOLECULE_NAME_PROPERTY_KEY));
+            i++;
+        }
+
+        /*
+        Expected output:    50 parsable lines
+                            1 invalid lines (header)
+        Test file's specifications:
+        - 51 lines, 50 with structures, 1 header line
+        - ID second in line
+        - used separator: " "
+        - multiple garbage columns after the first 2
+         */
+        tmpURL = this.getClass().getResource("SMILESTestFileSix.smi");
+        tmpFormat = DynamicSMILESFileReader.detectFormat(Paths.get(tmpURL.toURI()).toFile());
+        Assertions.assertTrue(tmpFormat.hasHeaderLine());
+        Assertions.assertEquals(1, tmpFormat.getIDColumnPosition());
+        Assertions.assertEquals(0, tmpFormat.getSMILESCodeColumnPosition());
+        Assertions.assertTrue(tmpFormat.hasIDColumn());
+        Assertions.assertEquals(' ', tmpFormat.getSeparatorChar());
+        tmpMolSet = DynamicSMILESFileReader.readFile(Paths.get(tmpURL.toURI()).toFile(), tmpFormat);
+        Assertions.assertEquals(50, tmpMolSet.getAtomContainerCount());
+        Assertions.assertEquals("CNP0000001", tmpMolSet.getAtomContainer(0).getProperty(Importer.MOLECULE_NAME_PROPERTY_KEY));
+    }
```

Comment:
I think it would be better to split this test (above) into several smaller tests

Reply:
Fixed in https://github.com/FelixBaensch/MORTAR/pull/31/commits/7145deca0c6fbc07c98a2355dadc419c5cb69b57